presto-memory/src/test/java/com/facebook/presto/plugin/memory/TestMemorySmoke.java

```diff
@@ -101,47 +97,46 @@ public void testCreateFilteredOutTable()
     public void testSelectFromEmptyTable()
             throws SQLException
     {
-        queryRunner.execute("CREATE TABLE test_select_empty AS SELECT * FROM tpch.tiny.nation WHERE nationkey > 1000");
+        computeActual("CREATE TABLE test_select_empty AS SELECT * FROM tpch.tiny.nation WHERE nationkey > 1000");
 
         assertQueryResult("SELECT count(*) FROM test_select_empty", 0L);
     }
 
     @Test
     public void testSelectSingleRow()
     {
-        assertQuery("SELECT * FROM nation WHERE nationkey = 1", "SELECT * FROM tpch.tiny.nation WHERE nationkey = 1");
+        assertQueryEquals("SELECT * FROM nation WHERE nationkey = 1", "SELECT * FROM tpch.tiny.nation WHERE nationkey = 1");
     }
 
     @Test
     public void testSelectColumnsSubset()
             throws SQLException
     {
-        assertQuery("SELECT nationkey, regionkey FROM nation ORDER BY nationkey", "SELECT nationkey, regionkey FROM tpch.tiny.nation ORDER BY nationkey");
+        assertQueryEquals("SELECT nationkey, regionkey FROM nation ORDER BY nationkey", "SELECT nationkey, regionkey FROM tpch.tiny.nation ORDER BY nationkey");
     }
 
     @Test
     public void testCreateTableInNonDefaultSchema()
     {
-        queryRunner.execute(format("CREATE SCHEMA %s.schema1", CATALOG));
-        queryRunner.execute(format("CREATE SCHEMA %s.schema2", CATALOG));
+        computeActual(format("CREATE SCHEMA %s.schema1", CATALOG));
+        computeActual(format("CREATE SCHEMA %s.schema2", CATALOG));
 
         assertQueryResult(format("SHOW SCHEMAS FROM %s", CATALOG), "default", "information_schema", "schema1", "schema2");
-
-        queryRunner.execute(format("CREATE TABLE %s.schema1.nation AS SELECT * FROM tpch.tiny.nation WHERE nationkey %% 2 = 0", CATALOG));
-        queryRunner.execute(format("CREATE TABLE %s.schema2.nation AS SELECT * FROM tpch.tiny.nation WHERE nationkey %% 2 = 1", CATALOG));
+        computeActual(format("CREATE TABLE %s.schema1.nation AS SELECT * FROM tpch.tiny.nation WHERE nationkey %% 2 = 0", CATALOG));
+        computeActual(format("CREATE TABLE %s.schema2.nation AS SELECT * FROM tpch.tiny.nation WHERE nationkey %% 2 = 1", CATALOG));
 
         assertQueryResult(format("SELECT count(*) FROM %s.schema1.nation", CATALOG), 13L);
         assertQueryResult(format("SELECT count(*) FROM %s.schema2.nation", CATALOG), 12L);
     }
 
     private List<QualifiedObjectName> listMemoryTables()
     {
-        return queryRunner.listTables(queryRunner.getDefaultSession(), "memory", "default");
+        return getQueryRunner().listTables(getSession(), "memory", "default");
     }
 
     private void assertQueryResult(String sql, Object... expected)
     {
-        MaterializedResult rows = queryRunner.execute(sql);
```

Comment:
You can replace this method with
```java
assertEquals(computeActual(sql).getOnlyColumnAsSet(), ImmutableSet.copyOf(expected));

Reply:
Replacing the method with your suggestion gives a cast error because `getOnlyColumnAsSet` expects the output to be a String.
```
java.lang.ClassCastException: java.lang.Long cannot be cast to java.lang.String
	at com.facebook.presto.testing.MaterializedResult.lambda$getOnlyColumnAsSet$0(MaterializedResult.java:196)
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
	at java.util.Collections$2.tryAdvance(Collections.java:4717)
	at java.util.Collections$2.forEachRemaining(Collections.java:4725)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499)
	at com.facebook.presto.testing.MaterializedResult.getOnlyColumnAsSet(MaterializedResult.java:197)
	at com.facebook.presto.plugin.memory.TestMemorySmoke.assertQueryResult(TestMemorySmoke.java:157)

```

This seems to work, though it appears a bit...strange. :D 

```
    private <T> void assertQueryResult(@Language("SQL") String sql, T... expected)
    {
        assertEquals(computeActual(sql).<T>getOnlyColumnAsSet(), ImmutableList.copyOf(expected));
    }

```
In MaterializedResult:
```
    public <T> Set<T> getOnlyColumnAsSet()
    {
        checkState(types.size() == 1, "result set must have exactly one column");
        return rows.stream()
                .map(row -> (T) row.getField(0))
                .collect(toImmutableSet());
    }
```

Reply:
This is clever, but it's not type safe. With the old way, the cast to `String` happened at runtime, so if the query returned a non-string, it would fail with `ClassCastException` as you found. With a cast to a generic type, there is no runtime cast, since the types are [erased](https://docs.oracle.com/javase/tutorial/java/generics/erasure.html) at compile time. So you can get back a `Set<Integer>`, but it might actually contain `String`. That can cause serious problems that are really hard to debug.

One solution that's sometimes used is to pass in the type:
```java
public <T> Set<T> getSomething(Class<T> type)
{
     return type.cast(value);
}
```

In this case, we should just leave the old method. I didn't realize they were using non-string values.

Reply:
I see. Got it, thank you for the explanation David!